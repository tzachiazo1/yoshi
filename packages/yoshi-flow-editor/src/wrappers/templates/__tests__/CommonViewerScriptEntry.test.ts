import {
  OOI_WIDGET_COMPONENT_TYPE,
  PLATFORM_WIDGET_COMPONENT_TYPE,
} from 'yoshi-flow-editor-runtime/build/constants';
import commonViewerScriptEntry from '../CommonViewerScriptEntry';

describe('CommonViewerScriptEntry template', () => {
  it('generates correct template with a single controller', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      appName: 'someapp',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/button/controller.ts',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with a single controller and without viewerEntryFileName', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      appName: 'someApp',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/button/controller.ts',
          componentName: 'SomeComp',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: null,
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with multiple controllers', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      appName: 'someapp',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '567',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with controllerId override', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      appName: 'someapp',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '123',
          widgetType: PLATFORM_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
        {
          controllerFileName: 'project/src/components/todo/controller.ts',
          id: '567',
          controllerId: '09876',
          widgetType: PLATFORM_WIDGET_COMPONENT_TYPE,
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template w/o controllers', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: null,
      appName: 'someApp',
      controllersMeta: [],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });

  it('generates correct template with sentry', () => {
    const generateControllerEntryContent = commonViewerScriptEntry({
      viewerScriptWrapperPath:
        'yoshi-flow-editor-runtime/build/viewerScript.js',
      sentryConfig: {
        DSN: 'https:xxx@123',
        teamName: 'team-name',
        projectName: 'project-name',
        id: 'xxx',
      },
      appName: 'someApp',
      controllersMeta: [
        {
          controllerFileName: 'project/src/components/button/controller.ts',
          id: '123',
          widgetType: OOI_WIDGET_COMPONENT_TYPE,
          controllerId: 'CONTROLLER_ID',
          componentName: 'SomeWidget',
        },
      ],
      experimentsConfig: {
        scope: 'test-scope',
      },
      viewerEntryFileName: 'project/src/app.ts',
    });

    expect(generateControllerEntryContent).toMatchSnapshot();
  });
});
