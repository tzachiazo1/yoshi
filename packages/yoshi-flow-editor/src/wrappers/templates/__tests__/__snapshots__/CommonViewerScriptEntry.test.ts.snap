// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonViewerScriptEntry template generates correct template w/o controllers 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {
    scope: 'test-scope'
  };

  export const initAppForPage = initAppForPageWrapper(importedApp.initAppForPage, sentryConfig, experimentsConfig);
  export const createControllers = createControllersWithDescriptors([]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with a single controller 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/button/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {
    scope: 'test-scope'
  };

  export const initAppForPage = initAppForPageWrapper(importedApp.initAppForPage, sentryConfig, experimentsConfig);
  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          id: \\"123\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with a single controller and without viewerEntryFileName 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/button/controller.ts';

  
  var importedApp = {};


  var sentryConfig = null;

  var experimentsConfig = {
    scope: 'test-scope'
  };

  
  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          id: \\"123\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with controllerId override 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/todo/controller.ts';
import controller1 from 'project/src/components/todo/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {
    scope: 'test-scope'
  };

  export const initAppForPage = initAppForPageWrapper(importedApp.initAppForPage, sentryConfig, experimentsConfig);
  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"STUDIO_WIDGET\\",
          id: \\"123\\" }, { method: controller1,
          widgetType: \\"STUDIO_WIDGET\\",
          id: \\"09876\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with multiple controllers 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/todo/controller.ts';
import controller1 from 'project/src/components/todo/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = null;

  var experimentsConfig = {
    scope: 'test-scope'
  };

  export const initAppForPage = initAppForPageWrapper(importedApp.initAppForPage, sentryConfig, experimentsConfig);
  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          id: \\"123\\" }, { method: controller1,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          id: \\"567\\" }]);
"
`;

exports[`CommonViewerScriptEntry template generates correct template with sentry 1`] = `
"
  import {createControllersWithDescriptors, initAppForPageWrapper} from 'yoshi-flow-editor-runtime/build/viewerScript.js';
  
  import controller0 from 'project/src/components/button/controller.ts';

  
  import * as viewerApp from 'project/src/app.ts';
    var importedApp = viewerApp;


  var sentryConfig = {
      DSN: 'https:xxx@123',
      id: 'xxx',
      projectName: 'project-name',
      teamName: 'team-name',
    };

  var experimentsConfig = {
    scope: 'test-scope'
  };

  export const initAppForPage = initAppForPageWrapper(importedApp.initAppForPage, sentryConfig, experimentsConfig);
  export const createControllers = createControllersWithDescriptors([{ method: controller0,
          widgetType: \\"WIDGET_OUT_OF_IFRAME\\",
          id: \\"123\\" }]);
"
`;
